### PR TITLE
Change "showModuleAction" variables to "allowModuleAction"

### LINKED ISSUE #1777: Can't overwrite {% if showBlogDelete %}
## Type
- Non critical bug
## Problem description

> If for any reason you want to overwrite the default "showBlogDelete", this seems to be not possible anymore. Which I think was possible in older fork cms 3.9.\* versions (which used SpoonTemplate).
## Steps to reproduce

> You can try to add the following in your parse() method in the Edit.php action class of the Blog or ContentBlocks module:

``` php
$this->tpl->assign('showBlogDelete', false);
```
## Expected behavior / Proposed solutions

> Expected behavior is that the delete button is gone. But it is still available...

### PR DESCRIPTION
**Pull request by jeroendesloovere**
Fixes #1777

### CODE REVIEW

**Comment by carakas** [comment]
@jeroendesloovere can you fix the errors that showed up on the CI

**Comment by jeroendesloovere** [comment]
@carakas those errors aren't mine?

**Comment by carakas** [comment]
@jeroendesloovere I think they are, maybe your code has side effects.

Because master doesn't have these errors

**Comment by jeroendesloovere** [comment]
Discovered a side effect when clicking to "Modules".
schermafbeelding 2016-11-07 om 13 20 09

**Comment by jeroendesloovere** [comment]
When you visit Dashboard, the BaseTwigTemplate is instantiated three times.
So that makes my code useless.

**Comment by jeroendesloovere** [comment]
This should not be possible. Since they should be run in a sandbox.
I'm investigating this further.

**Comment by jeroendesloovere** [comment]
The problem lies at the following rule:
https://github.com/forkcms/forkcms/blob/master/src/Backend/Core/Engine/TwigTemplate.php#L39

We can solve this by removing the if statement and the function parameter and just saying:
```
Model::getContainer()->set('template', $this);
```
But don't know if that gives further problems...
Since template is begin used at more places:
schermafbeelding 2016-11-07 om 13 51 43

File: src/Backend/Modules/Pages/Actions/Edit.php [hunk 1]
```diff
@@ -541,9 +541,6 @@ protected function parse()
             $showDelete = false;
         }
 
-        // show delete button
-        $this->tpl->assign('showPagesDelete', $showDelete);
```
Comment (carakas) [inline src/Backend/Modules/Pages/Actions/Edit.php]: this isn't the fix, you should use a different template variable name for this. you are changing the intended implementation

File: src/Backend/Modules/Blog/Layout/Templates/EditCategory.html.twig [hunk 2]
```diff
@@ -63,7 +63,7 @@
           </button>
         </div>
       </div>
-      {% if showBlogDeleteCategory %}
+      {% if showBlogDeleteCategory and allowBlogDeleteCategory %}
```
Comment (carakas) [inline src/Backend/Modules/Blog/Layout/Templates/EditCategory.html.twig]: isn't the showBlogDeleteCategory variable already checked in the BackendBlogModel?

please read the code very well when making changes like this

Comment (jeroendesloovere) [inline src/Backend/Modules/Blog/Layout/Templates/EditCategory.html.twig]: @carakas This is something different.
See: https://github.com/forkcms/forkcms/pull/1828/files#diff-8f7f5362034515c7d4353de51b90b63aR83

**Pull request merged**